staff/acct/sorry/sorry

```diff
@@ -1,156 +1,183 @@
-chmod 500 "$userdir"
-
-echo ""
-echo ""
-echo ""
-echo "Final system check"
-ldapsearch -x uid="$sorriedUser"
-ldapsearch -x cn=sorry | tail
-finger -m "$sorriedUser"
-ls -la "$userdir"
-ls -ld "$httpdir"
-
-# logging
-echo "$(/bin/date) - $rootstaffer $sorriedUser" >> /opt/acct/sorry.log
-
-# Notify user by email if email address is available.
-email_from='Open Computing Facility <help@ocf.berkeley.edu>'
-email_subject='[OCF] Account disabled'
-email_to="$(ldapsearch "uid=${sorriedUser}" mail | grep ^mail | cut -d' ' -f2)"
-[ -n "$email_to" ] && \
-  mail -a "From: ${email_from}" -s "$email_subject" "$email_to" <<EOF
+#!/usr/bin/env python3
+# Script for sorrying OCF user accounts rewritten in python3
+import argparse
+import datetime
+import grp
+import os
+import shutil
+import subprocess
+import sys
+
+import ocflib.account.manage
+import ocflib.account.search
+import ocflib.account.utils
+import ocflib.misc.mail
+
+reason_path = '/opt/share/utils/staff/acct/sorry/'
+sorry_shell = '/opt/share/utils/bin/sorried'
+
+
+def find_sorry_file(file_path):
+    # Own sorry file
+    if (os.path.isfile(file_path)):
+        return file_path
+
+    # Default sorry files
+    if (os.path.isfile(reason_path + file_path)):
+        return reason_path + file_path
+
+    raise FileNotFoundError("FATAL: Sorry file \"{}\" does not exist. Exiting.".format(file_path))
+
+
+def sorry_user(username, sorry_file):
+    attributes = ocflib.account.search.user_attrs(username)
+    if attributes is None:
+        raise ValueError('User not found.')
+
+    rootstaffer = real_user()
+    kerberos_credentials(rootstaffer)
+
+    user_dir = attributes['homeDirectory']
+
+    # Workaround for ldap3 1.x.x machines
+    if isinstance(user_dir, list):
+        user_dir = user_dir[0]
+
+    print('Copying sorry file and making .oldshell file')
+
+    if not (os.path.isfile(user_dir + '/.oldshell')):
+        oldshell = attributes['loginShell']
+        # Workaround for ldap3 1.x.x machines
+        if isinstance(oldshell, list):
+            oldshell = user_dir[0]
+
+        oldshell_file = open(user_dir + '/.oldshell', 'w')
+        oldshell_file.write(oldshell)
+        oldshell_file.close()
+
+    shutil.copy2(sorry_file, user_dir + '/.sorry')
+    os.chmod(user_dir + '/.sorry', 0o400)
+    shutil.chown(user_dir + '/.sorry', user=username, group='ocf')
+
+    print("Changing user's shell to sorry shell")
+
+    ocflib.infra.ldap.modify_ldap_entry(ocflib.account.utils.dn_for_username(username),
+                                        {'loginShell': sorry_shell, 'gidNumber': grp.getgrnam('sorry').gr_gid})
+
+    http_dir = '/services/http/users/{0}/{1}'.format(username[0], username)
+    print('Changing permissions on httpdir to 000')
+    if (os.path.isdir(http_dir)):
+        os.chmod(http_dir, 0o000)
+
+    print('Changing permissions on home directory to 500')
+    os.chmod(user_dir, 0o500)
+
+    # WTF is final system check for?
```

Comment:
Honestly I think it's just to show some debugging information since this bash script is not the most stable and often has errors as it goes, so that's why that part is there and it can probably be removed on this python version.